crates/genie-lang/src/lib.rs

```diff
@@ -407,7 +444,8 @@ impl LangFile {
 
     /// Writes this language file to an output writer using the ini format.
     pub fn write_to_ini<W: Write>(&self, output: &mut W) -> io::Result<()> {
-        for (id, string) in self.iter() {
+        // TODO warning if there are string ids
+        for (id, string) in self.iter().filter(|(id, _)| id.is_numeric()) {
```

Comment:
I was thinking about warnings for genie-scx as well, because conversions from HD to WK and such can lose information too. My idea was returning a custom warning enum in `io::Result<Option<Warning>>` (or `io::Result<Vec<Warning>>` if there can be multiple). I've not really seen that be done in any other crate before, but it seems quite sensible. Did you have any (other) implementation ideas for this already?

Reply:
Specifically for the ini writing, since the `LangFile` struct is not mutated by that function, it might suffice just to return a `io::Result<bool>` indicating true/false whether any strings were skipped during the writing. If they were, then the struct can still be iterated over/filtered to obtain the named strings that were skipped.

Some type of general purpose conversion warning might belong more appropriately to the conversion application rather than to the libraries.

But for the more general issue of how to represent a warning using the Result type, I'm not sure. It might make sense to use the option type if there is some data associated with a warning. Something like:

```rust
if let Some(warning) = foo()? {
    // handle warning
};
```

I'm not actually sure how best to use a Result type to represent a "Partial success". Perhaps a new 3-tuple type would be better, e.g. a `Result<Ok, Partial, Err>`. But this could be implemented using a Result like `Result<(Ok, Partial), Err>` or `Result<Ok, (Partial, Err)>`.

Reply:
One example is the `read` method in `std::io::Read`, which uses the return value to return the number of bytes that were read: [https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read](https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read)

Reply:
I quite like `Result<(Ok, Partial), Err>`!